### PR TITLE
Add tests for SectionList

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by sccolbert** [comment]
Please resubmit without the generated /dist files

File: tests/collections/testsectionlist.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) 2015, Phosphor Contributors
+|
+| Distributed under the terms of the BSD 3-Clause License.
+|
+| The full license is in the file LICENSE, distributed with this software.
+|----------------------------------------------------------------------------*/
+module tests {
+
+  import SectionList = phosphor.collections.SectionList;
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: remove leading spaces

File: tests/collections/testsectionlist.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+
+
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: two blank lines after imports

File: tests/collections/testsectionlist.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+describe('phosphor.collections - sectionlist', () => {
+
+  describe('insert()', () => {
+
+    it('should return true on success', () => {
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: this description is not reflective of the test

File: tests/collections/testsectionlist.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      var obj = new SectionList();
+      expect(obj.size).to.be(0);
+      expect(obj.count).to.be(0);
+      obj.insert(0, 100, 10);
+      expect(obj.size).to.be(1000);
+      expect(obj.count).to.be(100);
+    });
+
+    it('should create heterogenous list', () => {
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: 'heterogeneous' typically refers to types, maybe 'list with variable sized sections' is more appropriate

File: tests/collections/testsectionlist.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      obj.insert(10, 100, 20);
+      expect(obj.size).to.be(3000);
+      expect(obj.count).to.be(200);
+      expect(obj.sizeOf(0)).to.be(10);
+      expect(obj.sizeOf(10)).to.be(20);
+      expect(obj.sizeOf(199)).to.be(10);
+      expect(obj.offsetOf(20)).to.be(300);
+    });
+
+    it('should create another heterogenous list', () => {
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: ditto

File: tests/collections/testsectionlist.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      expect(obj.count).to.be(550);
+      expect(obj.sizeOf(0)).to.be(5);
+      expect(obj.sizeOf(549)).to.be(25);
+      expect(obj.sizeOf(125)).to.be(20);
+      expect(obj.sizeOf(225)).to.be(30);
+      expect(obj.sizeOf(300)).to.be(40);
+      expect(obj.sizeOf(475)).to.be(10);
+    });
+
+    it('should insert at 0', () => {
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: 'should insert at `list.count` when index is `>= list.count`', or a more general test which tests all of the parameter conditions described in the API docs.

File: tests/collections/testsectionlist.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+      var obj = new SectionList();
+      obj.insert(0, 100, 10);
+      obj.insert(10, 100, 20);
+      obj.remove(125, 50);
+      obj.remove(75, 50);
+      obj.remove(25, 50);
+      obj.remove(0, 50);
+      expect(obj.count).to.be(0);
+    });
+
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: should also have explicit tests for 'count', 'size', 'indexOf', 'offsetOf', and 'sizeOf'. Even though there is overlap, it's better to have explicit tests for each part of the public API

File: tests/collections/testsectionlist.ts [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+      var obj = new SectionList();
+      obj.insert(0, 50, 20);
+      obj.insert(-10, 75, 10);
+      expect(obj.size).to.be(1750);
+      expect(obj.count).to.be(125);
+      expect(obj.sizeOf(0)).to.be(20);
+      expect(obj.sizeOf(39)).to.be(20);
+      expect(obj.sizeOf(40)).to.be(10);
+    });
+
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: one blank line

File: tests/collections/testsectionlist.ts [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+  });
+
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: two blank lines

File: tests/collections/testsectionlist.ts [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+      obj.insert(0, 100, 10);
+      obj.insert(100, 30, 20);
+      obj.remove(-20, 10);
+      expect(obj.count).to.be(120);
+      expect(obj.sizeOf(119)).to.be(20);
+      obj.remove(-30, 10);
+      expect(obj.count).to.be(110);
+      expect(obj.sizeOf(109)).to.be(20);
+    });
+
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: 1 blank

File: tests/collections/testsectionlist.ts [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+
+  });
+
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: 2 blanks

File: tests/collections/testsectionlist.ts [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+      expect(obj.sizeOf(50)).to.be(20);
+      expect(obj.sizeOf(40)).to.be(10);
+      expect(obj.sizeOf(60)).to.be(10);
+    });
+
+
+    it('should resize the end of the list', () => {
+      var obj = new SectionList();
+      obj.insert(0, 100, 10);
+      obj.resize(90, 20, 20);  // extends past the end of the list
```
Comment (sccolbert) [inline tests/collections/testsectionlist.ts]: can you also assert the list count here? it should not change even though the resize extends past the end

**Comment by blink1073** [comment]
Superseded by #42.

**Pull request closed without merging**